msvc-specific.cmake

```diff
@@ -0,0 +1,122 @@
+  endif()
+ENDMACRO(GET_MSVC_VERSION)
+
+
+# REQUEST_MINIMUM_MSVC_VERSION(VER)
+# ---------------------------------
+#
+# Return a fatal error if the current MSVC version is strictly lower than VER
+# or when no MSVC compiler or an unknown version of it is used.
+#
+# A reason for the version to be unknown is that it is newer than the versions
+# recognize by the above macro GET_MSVC_VERSION. In this case, please update 
+# the macro and its documentation.
+#
+MACRO(REQUEST_MINIMUM_MSVC_VERSION)
+  if(NOT ${ARGC} GREATER 0)
+    message("REQUEST_MINIMUM_MSVC_VERSION requires an argument" FATAL_ERROR)
+  endif()
+  GET_MSVC_VERSION()
+  if (${MSVC_VERSION})
+    if (${MSVC_VERSION} STRGREATER ${ARGV0})
+    else()
+	  message("Minimum MSVC version required is " ${ARGV0} FATAL_ERROR)
+    endif()
+  else()
+    message("Invalid MSVC version" FATAL_ERROR)
+  endif(${MSVC_VERSION})
+ENDMACRO(REQUEST_MINIMUM_MSVC_VERSION)
+
+# GENERATE_DOT_USER_FILE(NAME)
+# ----------------------------
+#
+# Generate the file NAME.vcxproj.user alongside the project file NAME.vcxproj
+# that sets up the path for a debugging session by adding to %PATH% the 
+# location of the libraries generated by the solution, i.e. it sets the field
+# Configuration Properties > Debuging > Environment to 
+# PATH=$(SolutionDir)\src\$(Configuration);%PATH%
+# This is intended to be used for the test projects, so that they can be 
+# launched from within the Visual Studio IDE by pressing F5.
+#
+MACRO(GENERATE_DOT_USER_FILE)
+  if(MSVC)
+    REQUEST_MINIMUM_MSVC_VERSION("10.0")
+	
+    if(NOT ${ARGC} GREATER 0)
+      message("GENERATE_DOT_USER_FILE requires to specify the project name" FATAL_ERROR)
+    endif()
+    
+    if(${CMAKE_CL_64})
+      set(MSVC_COMPILER_VERSION "x64")
+    else()
+      set(MSVC_COMPILER_VERSION "Win32")
+    endif()
+	
+	GET_MSVC_VERSION()
+	set(DOT_USER_FILE_PATH ${CMAKE_CURRENT_BINARY_DIR}/${ARGV0}.vcxproj.user)
+	
+	file(WRITE ${DOT_USER_FILE_PATH} <?xml " " version=\"1.0\" " " encoding=\"utf-8\"?>\n)
+	file(APPEND ${DOT_USER_FILE_PATH} <Project " " ToolsVersion=\"14.0\" " " xmlns=\"http://schemas.microsoft.com/developer/msbuild/2003\">\n)
+	FOREACH(config ${CMAKE_CONFIGURATION_TYPES})
+      file(APPEND ${DOT_USER_FILE_PATH} "  " <PropertyGroup " " Condition=\"'\$(Configuration) | \$(Platform) ' =='${config}|${MSVC_COMPILER_VERSION}'\">\n)
+	  file(APPEND ${DOT_USER_FILE_PATH} "    " <LocalDebuggerEnvironment>PATH=\$(SolutionDir) \\ src \\ \$ (Configuration) \; %PATH%</LocalDebuggerEnvironment>\n)
+	  file(APPEND ${DOT_USER_FILE_PATH} "    " <DebuggerFlavor>WindowsLocalDebugger</DebuggerFlavor>\n)
+	  file(APPEND ${DOT_USER_FILE_PATH} "  " </PropertyGroup>\n)
+    ENDFOREACH()
+	file(APPEND ${DOT_USER_FILE_PATH} </Project>)
+  else(MSVC)
+    message("This macro is intended for MSVC only, skipping .user file generation")
+  endif(MSVC)
+ENDMACRO(GENERATE_DOT_USER_FILE)
```

Comment:
This is great but you could use configure_file to generate the .user file here (several modules in jrl-cmakemodules make the assumption that the module is located under the `cmake` folder for this)